### DISCUSSION TITLE
Easy way to get motif.names from motif.ID?

### DISCUSSION THREAD

**Discussion by franciscogrisanti** [stuart-lab/signac#673] [Q&A] [answered]
Hi  @timoast !

I have a list of differential motif activity 

<img width="330" alt="image" src="https://user-images.githubusercontent.com/35009018/122070560-b12c9700-cdbb-11eb-94ab-87326e32ee6e.png">

Is there an easy way to pass from those motif ID to motif names? Thanks in advance!

Best regards,

**Comment by paulasoler** [comment]
Hi @franciscogrisanti!

If you are using the Jaspar database, you can do something like this:

motifs_names <- sapply(row.names(df), function(x) {name(getMatrixByID(JASPAR2020, ID = x))})

Best regards,

**Comment by timoast** [comment, marked as answer]
You can use the `ConvertMotifID()` function

**Comment by franciscogrisanti** [comment]
Thank you both for the prompt response!

@timoast I don't know why ConvertMotifID() didn't work for me. Here is my sample code

`ConvertMotifID(obj,
               name=rownames(obj@assays$chromvar@data),
               assay='chromvar')

ConvertMotifID(obj@assays$chromvar,
               names=rownames(obj@assays$chromvar@data),
               assay='chromvar')

`


@paulasoler This seems to work out! There are some motifs that doesn't appear to be on the database. I don't know how is that possible. I'm pretty sure I'm using Jaspar2020. I had to modify a little your code so it can return NULL when it doesnt find the motif


`get_motif_name <- function(x){
    name(getMatrixByID(JASPAR2020, ID = x))
}

get_motif_name_w_error <- function (x) {
  out <- tryCatch(get_motif_name(x), error = function(e) NULL)
  return(out)
}
                  
motif.names <- sapply(row.names(differential.activity),get_motif_name_w_error)`

**Reply by timoast** [reply]
What error did you get? My guess is you need to set the `id` parameter rather than `name`

**Comment by coopershawna** [comment]
I'm having trouble with this too.
ids <- c("MA0030.1", "MA0031.1", "MA0051.1", "MA0057.1")
names <- ConvertMotifID(object = chromvar.average.integrated_zones, id = ids, assay = 'chromvar')
names

output: 
> names
           CV_Control CV_Alcohol ML_Control ML_Alcohol PV_Control PV_Alcohol
MA0030.1  0.343086674  0.7403037 0.11994472  0.4385891  0.3187846  0.7245411
MA0031.1  0.503937996  0.5562221 0.28058035  0.3714607  0.3515111  0.6214056
MA0051.1  0.087940582  0.5520646 0.08573117  0.3365035  1.1584318  1.1828982
MA0057.1 -0.001884508  0.5282958 0.04887257  0.4626327  0.2784042  0.8601793

I've run my entire analysis, but the heatmaps are coming out with the matrix id labels. I need to convert to the common names. 
It is recognizing the matrix id and returning values, but isn't converting them. 

I also tried 
chromvar.average.integrated_zones.names <- ConvertMotifID(chromvar.average.integrated_zones, id = rownames(chromvar.average.integrated_zones@assays$chromvar@data), assay = 'chromvar')

and 
chromvar.average.integrated_zones <- ConvertMotifID(chromvar.average.integrated_zones, id = rownames(chromvar.average.integrated_zones@assays$chromvar@data), assay = 'chromvar')

Both just return a table with the matrix id and values.

I'd really appreciate some insight. Thank you!
